src/main/java/org/opensearch/geospatial/ip2geo/action/RestPutDatasourceAction.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.geospatial.ip2geo.action;
+
+import static org.opensearch.geospatial.shared.URLBuilder.URL_DELIMITER;
+import static org.opensearch.geospatial.shared.URLBuilder.getPluginURLPrefix;
+import static org.opensearch.rest.RestRequest.Method.PUT;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.geospatial.ip2geo.common.Ip2GeoSettings;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.action.RestToXContentListener;
+
+/**
+ * Rest handler for Ip2Geo datasource creation
+ */
+public class RestPutDatasourceAction extends BaseRestHandler {
```

Comment:
will it be better we rename this to RestPutDatasourceHandler? Action is mainly used for Transport Actions and as this is a RestHandler it will be good if we have that distinction.

I understand we have older Rest handlers named as Action but that creates a lot of confusion. We can rename those classes also, as part of another PR it will be great.

Reply:
Other part of codes follows similar pattern. I would stay consistent with them.
`RestUploadGeoJSONAction`, `RestUploadStatsAction`

Reply:
I think its time to break that pattern and create a github issue to fix the older code.

Reply:
Done